odoo/addons/base/ir/ir_actions_report.py

```diff
@@ -576,19 +483,89 @@ def render_template(self, template, values=None):
         return view_obj.render_template(template, values)
 
     @api.model
-    def render_qweb_pdf(self, res_ids, html=None, data=None):
+    def _post_pdf(self, attachment_ids, pdf_content=None, res_ids=None):
+        '''Merge the existing attachments by adding one by one the content of the attachments
+        and then, we add the pdf_content if exists. Create the attachments for each record individually
+        if required.
+
+        :param attachment_ids: The retrieved attachments.
+        :param pdf_content: The pdf content newly generated by wkhtmltopdf.
+        :param res_ids: the ids of record to allow postprocessing.
+        :return: The pdf content of the merged pdf.
+        '''
+
+        def close_streams(streams):
+            for stream in streams:
+                try:
+                    stream.close()
+                except Exception:
+                    pass
+
+        # Check special case having only one record with existing attachment.
+        if len(attachment_ids) == 1 and not pdf_content:
+            return base64.decodestring(attachment_ids[0].datas)
+
+        # Append all attachments content and then, append pdf_content
+        writer = PdfFileWriter()
+        streams = []
+        # Decode each attachment and append
+        for attachment_id in attachment_ids:
+            content = base64.decodestring(attachment_id.datas)
+            stream = StringIO(content)
+            streams.append(stream)
+            reader = PdfFileReader(stream)
+            writer.appendPagesFromReader(reader)
+        # Append content of pdf if exists
+        if pdf_content:
+            stream = StringIO(pdf_content)
+            streams.append(stream)
+            reader = PdfFileReader(stream)
+            # Call the postprocess method for each record.
+            if res_ids and self.attachment_use and self.attachment:
+                # Build a record_map mapping id -> record
+                record_map = dict(
+                    [(r.id, r) for r in self.env[self.model].browse([res_id for res_id in res_ids if res_id])])
+                if len(res_ids) == 1 and res_ids[0] in record_map:
+                    # Only one record, so postprocess directly and append the whole pdf.
+                    self.postprocess_pdf_report(record_map[res_ids[0]], pdf_content)
```

Comment:
@smetl but if the pdf_content is a StringIO and the StringIO is added to the stream after the call to subprocess it's possible to support this functionality. Am I wrong?